src/_bentoml_sdk/method.py

```diff
@@ -107,12 +107,22 @@ def __get__(self: T, instance: None, owner: type) -> T: ...
     def __get__(self, instance: object, owner: type) -> t.Callable[P, R]: ...
 
     def __get__(self: T, instance: t.Any, owner: type) -> t.Callable[P, R] | T:
-        from pydantic.fields import FieldInfo
-        from pydantic_core import PydanticUndefined
-
         if instance is None:
             return self
 
+        local_caller = self._local_call(instance)
+
+        if proxy := getattr(instance, "__self_proxy__", None):
+            func = getattr(proxy, self.name)
+        else:
+            func = local_caller
+        func.local = local_caller  # type: ignore[attr-defined]
+        return func
+
+    def _local_call(self, instance: t.Any) -> t.Callable[P, R]:
+        from pydantic.fields import FieldInfo
```

Comment:
Why the import here? I assume this is a side effect in L113?

Reply:
it's used by L137

Reply:
oh i see.